content/blog/2023-12-14-private-repos.md

```diff
@@ -0,0 +1,70 @@
+---
+ date: 2023-12-14
+ title: Private Repositories
+ summary: Initial support for private repositories
+ author: Michele Baldessari
+ blog_tags:
+ - patterns
+ - announce
+ aliases: /2023/12/14/private-repositories/
+---
+
+We're excited to announce that support for private repositories is now
+available. This feature is accessible when using VP operator version 0.0.36 or
+higher, in conjunction with the latest common/ clustergroup 0.8.2 chart. With
+this update, you can deploy patterns from git repositories that are either
+password-protected or secured with an SSH key.
+
+To enable this feature, follow these steps:
+
+1. Create a Secret for Repository Access: Generate a secret that holds the
+   credentials for accessing your repository. This secret should be formatted
+   according to ArgoCD's guidelines, which you can find [here](https://argo-cd.readthedocs.io/en/stable/operator-manual/declarative-setup/#repositories).
+   For instance, your secret might look like this:
+   ```yaml
+   apiVersion: v1
+   kind: Secret
+   metadata:
+     name: private-repo
+     namespace: openshift-operators
+     labels:
+       argocd.argoproj.io/secret-type: repository
+   stringData:
+     type: git
+     url: git@github.com:mbaldessari/mcg-private.git
+     sshPrivateKey: |
+       -----BEGIN OPENSSH PRIVATE KEY-----
+       a3...
+       ...
+       ...
+       -----END OPENSSH PRIVATE KEY-----
+   ```
+2. Deploy the Pattern with the Secret: Point your pattern's Custom Resource to
+   the secret you created in the first step. Ensure that both tokenSecret and
```

Comment:
Is there (or should there be) a way to do this from the command line?
If so, should we mention it here?

Reply:
Good point, currently there isn't a way that just works out of the box with `make install`, I can add one to common. Let me take a look

Reply:
Mmh actually doing it via `make install` is a bit more involved than I had hoped. Doing it via UI and via a pattern CR is trivial. The make install part is a bit meh when doing ssh-protected private repos because we have logic in the makefile to always convert the git url to https. I am not too fond of suddenly changing that logic. I pushed https://github.com/validatedpatterns/common/pull/420 to get started with this.

Reply:
How about https://github.com/validatedpatterns/common/pull/421 ?